### ISSUE TITLE
externalTrafficPolicy: Local only work 100% when replica=1

### ISSUE THREAD

**Issue by ntcong** [google/metallb#333]
**Is this a bug report or a feature request?**:
bug report

**What happened**:
I exposed a nginx ingress with a LoadBalancer and externalTrafficPolicy: Local
With externalTrafficPolicy: Cluster it works 100% of the time
WIth externalTrafficPolicy: Local, with replica=2, connection timeout on 50% of the request, 25% with replica=4

**What you expected to happen**:
No connection timeout

**How to reproduce it (as minimally and precisely as possible)**:
````
~ helm install stable/nginx-ingress --name ingress 
~ kubectl get svc/ingress-nginx-ingress-controller
NAME                               TYPE           CLUSTER-IP      EXTERNAL-IP     PORT(S)                      AGE
ingress-nginx-ingress-controller   LoadBalancer   10.233.12.169   external_ip   80:30080/TCP,443:30443/TCP   153d
````
curl 20 time
````
 ~  (for i in {1..20}; do curl http://external_ip -s --connect-timeout 0.1&& echo; done) | grep 404 | wc -l
20
````
set nginx-ingress replica to 2
````
 ~  (for i in {1..20}; do curl http://external_ip -s --connect-timeout 0.1&& echo; done) | grep 404 | wc -l
10
````

**Anything else we need to know?**:
- I setup my cluster with kubespray and calico. Kube-proxy running in ipvs mode
- 3 master and 4 node
- Traffic always go to 1 node even in 4 replica mode, but the node only response 1 for each 4 requests
- I'm using metallb with a /28 range in BGP mode.
- My problem looks exactly like https://github.com/google/metallb/issues/288, but with Local instead of Cluster

**Environment**:
- MetalLB version: helm metallb-0.8.0
- Kubernetes version: 1.12.1
- BGP router type/version: it's a juniper router, not sure about version
- OS (e.g. from /etc/os-release): Ubuntu 16.04.4 LTS
- Kernel (e.g. `uname -a`): Linux k8s-node04 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by eknudtson** [comment]
Same issues here.

Traffic is being received on nodes with pods via BGP advertisements properly, but IPVS has multiple pod entries for the MetalLB IP's virtualserver and attempts to proxy to pods not on the host.  

Without normal SNAT done in `cluster` externaltrafficpolicy mode, the pod on the proxied-to host attempts to communicate back to the client directly, which fails because the client expects a response from the MetalLB service IP.

It looks like the solution is to have IPVS not list pods on other nodes for MetalLB service IPs.

**Comment by danderson** [comment]
MetalLB does nothing with IPVS, so that sounds like a bug in upstream Kubernetes - specifically, kube-proxy.

**Comment by uablrek** [comment]
Yes, it's a k8s bug; https://github.com/kubernetes/kubernetes/issues/71596

**Comment by eknudtson** [comment]
Looks like it's fixed in k8s 1.14 : https://github.com/kubernetes/kubernetes/pull/72432
